samples/suit/recovery/README.rst

```diff
@@ -10,41 +10,40 @@ SUIT recovery application
+The application uses the SMP protocol via Bluetooth LE and SUIT to perform the recovery.
+The main application which uses this recovery firmware must also be using Bluetooth LE.
 
 Building
 ********
 
-The recovery application needs to be compatible with the main application in terms of hardware configuration.
-To achieve this, appropriate devicetree overlay files from the main application must be passed when building the recovery application as ``EXTRA_DTC_OVERLAY_FILE``
-.
-These devicetree also need to specify a partition for the recovery application.
-See the overlay files for the smp_transfer sample as an example.
+In standard applications the recovery firmware is built as part of the main application build.
+The two firmwares need to be compatible in terms of hardware configuration  (especially the memory partitions).
+To achieve this, appropriate devicetree overlay files from the main application must be passed to the recovery application.
+This is done by adding ``recovery.overlay`` and ``recovery_hci_ipc.ovelay`` files in the main application sysbuild directory.
+The former file will be passed automatically to the recovery application image and the latter to the recovery radio image.
+These devicetree files need to define the ``cpuapp_recovery_partition`` and ``cpurad_recovery_partition`` nodes respectively.`
+As an example, see the files in the ``samples/suit/smp_transfer`` sample.
 
-For example, to build the recovery application as a recovery image for the smp transfer sample, run:
+Apart from creating the mentioned files, the ``SB_CONFIG_SUIT_BUILD_RECOVERY`` sysbuild configuration option must be set in the main application.
+This will cause the recovery firmware to be built automatically as part of the main application build.
 
-``west build -p --sysbuild -b nrf54h20dk/nrf54h20/cpuapp -- -DEXTRA_DTC_OVERLAY_FILE='../smp_transfer/sysbuild/recovery.overlay' -Dhci_ipc_EXTRA_DTC_OVERLAY_FILE='<absolute_path_to_smp_transfer_sample>/sysbuild/recovery_hci_ipc.overlay'``
+For example, to build the ``smp_transfer`` sample with the recovery firmware, run:
+
+``west build --sysbuild -b nrf54h20dk/nrf54h20/cpuapp -- -DFILE_SUFFIX=bt -DSB_CONFIG_SUIT_BUILD_RECOVERY=y``
 
 Flashing
 ********
 
-Currently it is not possible to flash the application using ``west flash`` .
-Instead, ``nrfutil`` should be used to flash the hex files for the application core image, radio core image and SUIT storage:
-
-``nrfutil device program --options chip_erase_mode=ERASE_NONE --firmware build/recovery/zephyr/suit_installed_envelopes_application_merged.hex``
-
-``nrfutil device program --options chip_erase_mode=ERASE_NONE --firmware build/recovery/zephyr/suit_installed_envelopes_radio_merged.hex``
+No additional steps are needed to flash the recovery firmware together with the main application firmware.
+Simply run:
 
-``nrfutil device program --options chip_erase_mode=ERASE_NONE --firmware build/recovery/zephyr/zephyr.hex``
+``west flash``
```

Comment:
Is this really all? I don't think this step works for me. I can only flash the main application but not the recovery.

Reply:
It does work. Do you use the rebased version?

Reply:
I didn't realize the applications were merged together, but the recover is indeed flashed.